client/src/pages/Profile.jsx

```diff
@@ -38,22 +43,52 @@ const Profile = () => {
 
 	const handleUserName = (e) => setUsername(e.target.value);
 
-	const handleUpdateProfile = async () => {
-		const data = {
-			email,
-			username,
-			aboutMe: aboutRef.current.value,
-			gender: genderRef.current.value,
-			age: Number(ageRef.current.value),
+	const handleImageUpload = () => {
+		const fileInput = document.createElement('input');
+		fileInput.type = 'file';
+		fileInput.accept = 'image/*';
+		fileInput.onchange = (e) => {
+			const file = e.target.files[0];
+			if (file) {
+				setImageFile(file);
+				const reader = new FileReader();
+				reader.onload = (event) => {
+					setUploadedImageUrl(event.target.result);
+				};
+				reader.readAsDataURL(file);
+			}
 		};
-
+		fileInput.click();
+	};
+	const handleEditProfile = () => {
+		handleImageUpload();
+	};
+	const handleUpdateProfile = async () => {
+		const formData = new FormData();
+		formData.append('email', email);
+		formData.append('username', username);
+		formData.append('aboutMe', aboutRef.current.value);
+		formData.append('gender', genderRef.current.value);
+		formData.append('age', Number(ageRef.current.value));
+
+		if (imageFile) {
+			formData.append('profileImage', imageFile);
+		}
 		try {
-			const response = await api.post('/profile', data);
+			const response = await api.post('/profile', formData);
```

Comment:
why use formdata?

Reply:
compatible with file uploads, and it allows to handle both regular form fields and file uploads in a single request.